src/components/organisms/NewProfileModal/components/ProfileModalFetchUser/useCurrentModalUser.tsx

```diff
@@ -0,0 +1,27 @@
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+import { currentModalUserSelector } from "utils/selectors";
+
+import { isLoaded, useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+
+const useConnectCurrentModalUser = (userId?: string) => {
+  useFirestoreConnect([
+    {
+      collection: "users",
+      doc: userId ?? "",
+      storeAs: "currentModalUser",
+    },
+  ]);
+};
```

Comment:
`useFirestoreConnect` can accept functions. If that function you can make `!userId` guard statement. This exact implementation can cause issues when userId is underfined

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2291/commits/a1bc6fb1274bf1267476d08be3f705cfdb35ca63